keras_nlp/samplers/sampler.py

```diff
@@ -358,22 +367,30 @@ def one_step(
                 return current_index, prompt, mask
             return [current_index, prompt, mask, cache]
```

Comment:
I'm seeing a funky error here locally, where tf.while_loop is complaining about tuple (the line above), vs list (here) structure for return types. Only with XLA.

Let's do tuple everywhere, so return `current_index, prompt, mask, cache` here.